Delicious/Scenes/Home/HomeViewModel.swift

```diff
@@ -14,14 +14,92 @@ struct HomeViewModel {
 // MARK: - ViewModelType
 extension HomeViewModel: ViewModelType {
     struct Input {
-
+        let loadTrigger: Driver<Void>
+        let reloadTrigger: Driver<Void>
+        let selectTrigger: Driver<RecipeType>
+        let selectItemTrigger: Driver<HomeTableViewItem>
+        let searchTrigger: Driver<Void>
     }
 
     struct Output {
-
+        let data: Driver<[HomeTableViewSection]>
+        let isLoading: Driver<Bool>
+        let isReloading: Driver<Bool>
+        let tableViewState: Driver<EmptyView.State>
+        let loadedError: Driver<Error>
+        let selected: Driver<Void>
+        let search: Driver<Void>
     }
 
-    func transform(_ input: Input) -> Output {
-        return Output()
+    func transform(_ input: HomeViewModel.Input) -> HomeViewModel.Output {
+        let error = ErrorTracker()
+
+        let getRecipes = getItem(
+            loadTrigger: input.loadTrigger,
+            reloadTrigger: input.reloadTrigger) { _ in
+            return self.useCase.getRecipes().trackError(error)
+        }
+
+        let recipe = getRecipes.item
+            .map { recipes -> [HomeTableViewSection] in
+                if recipes.isEmpty { return [] }
+                let data = recipes.split()
+                let featuredSection = HomeTableViewSection.featuredSection(
+                    item: .featuredItem(items: data.left)
+                )
+                let lastestSection = HomeTableViewSection.latestSection(
+                    items: data.right.map { HomeTableViewItem.latestItem(item: $0) }
+                )
+                return [featuredSection, lastestSection]
+            }
+
+        let loadedError = error.withLatestFrom(recipe) { error, _ in error }
+        
+        let state = Driver.merge(
+            error.map { EmptyView.State.error($0) },
+            recipe.map { $0.isEmpty ? EmptyView.State.empty(Constant.emptyMessage) : .normal }
+        ).distinctUntilChanged {
+            switch $1 {
+            case .error:
+                switch $0 {
+                case .error:
+                    return false
+                default:
+                    return true
+                }
+            default:
+                return false
+            }
+        }
+
+        let selected = input.selectTrigger
+            .do(onNext: { self.navigator.toInfomation(recipe: $0) })
```

Comment:
truyền thẳng function vào luôn nhé, tương tự với các cái bên dưới.
```suggestion
            .do(onNext: self.navigator.toInfomation(recipe:))
```